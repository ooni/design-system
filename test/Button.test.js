/* global describe, test, expect */
import React from 'react'
import { renderWithTheme } from './index'
import { Button } from '../'

describe('Button', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <Button />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('render with width prop', () => {
    const { container } = renderWithTheme(
      <Button width={1} />
    )
    expect(container.firstChild).toMatchSnapshot()
    // XXX for some reason only a shallow render makes this assertion true
    expect(container.firstChild).toHaveStyleRule('width', '100%')
  })

  test('render with margin prop', () => {
    const { container } = renderWithTheme(
      <Button m={2} />
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule('margin', '8px')
  })

  test('render with fontSize prop', () => {
    const { container } = renderWithTheme(
      <Button fontSize={2} />
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule('font-size', '22px')
  })

  test('render with inverted prop', () => {
    const { container } = renderWithTheme(
      <Button inverted />
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule('color', '#0588cb')
    expect(container.firstChild).toHaveStyleRule('background-color', '#f8f9fa')
  })

  test('render with hollow prop', () => {
    const { container } = renderWithTheme(
      <Button hollow />
    )
    expect(container.firstChild).toMatchSnapshot()
    //  expect(container.firstChild).toHaveStyleRule('background-color', 'transparent')
  })

  test('render with border prop (and hollow)', () => {
    const { container } = renderWithTheme(
      <Button hollow border={4} />
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule('border-width', '4px')
    expect(container.firstChild).toHaveStyleRule('border-style', 'solid')
    expect(container.firstChild).toHaveStyleRule('border-color', '#0588cb')
  })

})
