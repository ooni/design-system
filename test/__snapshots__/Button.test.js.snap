// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with border prop (and hollow) 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #0588cb;
  background-color: transparent;
  border-radius: 20px;
  border-width: 2px;
  border-style: solid;
  border-color: #0588cb;
  border-width: 4px;
  font-size: 16px;
}

.c0:hover:enabled {
  border-color: #37a6ed;
}

.c0:active:enabled {
  border-color: #0f77b8;
}

.c0:enabled {
  cursor: pointer;
}

<button
  class="c0"
  font-size="1"
/>
`;

exports[`Button render with fontSize prop 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #f8f9fa;
  background-color: #0588cb;
  border-radius: 20px;
  border-width: 2px;
  font-size: 22px;
}

.c0:hover:enabled {
  background-color: #37a6ed;
}

.c0:active:enabled {
  background-color: #0f77b8;
}

.c0:enabled {
  cursor: pointer;
}

<button
  class="c0"
  font-size="2"
/>
`;

exports[`Button render with hollow prop 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #0588cb;
  background-color: transparent;
  border-radius: 20px;
  border-width: 2px;
  border-style: solid;
  border-color: #0588cb;
  border-width: 2px;
  font-size: 16px;
}

.c0:hover:enabled {
  border-color: #37a6ed;
}

.c0:active:enabled {
  border-color: #0f77b8;
}

.c0:enabled {
  cursor: pointer;
}

<button
  class="c0"
  font-size="1"
/>
`;

exports[`Button render with inverted prop 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #0588cb;
  background-color: #f8f9fa;
  border-radius: 20px;
  border-width: 2px;
  font-size: 16px;
}

.c0:hover:enabled {
  background-color: #f1f3f5;
}

.c0:active:enabled {
  background-color: #e9ecef;
}

.c0:enabled {
  cursor: pointer;
}

<button
  class="c0"
  font-size="1"
/>
`;

exports[`Button render with margin prop 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #f8f9fa;
  background-color: #0588cb;
  border-radius: 20px;
  border-width: 2px;
  margin: 8px;
  font-size: 16px;
}

.c0:hover:enabled {
  background-color: #37a6ed;
}

.c0:active:enabled {
  background-color: #0f77b8;
}

.c0:enabled {
  cursor: pointer;
}

<button
  class="c0"
  font-size="1"
/>
`;

exports[`Button render with width prop 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #f8f9fa;
  background-color: #0588cb;
  border-radius: 20px;
  border-width: 2px;
  width: 100%;
  font-size: 16px;
}

.c0:hover:enabled {
  background-color: #37a6ed;
}

.c0:active:enabled {
  background-color: #0f77b8;
}

.c0:enabled {
  cursor: pointer;
}

<button
  class="c0"
  font-size="1"
  width="1"
/>
`;

exports[`Button renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #f8f9fa;
  background-color: #0588cb;
  border-radius: 20px;
  border-width: 2px;
  font-size: 16px;
}

.c0:hover:enabled {
  background-color: #37a6ed;
}

.c0:active:enabled {
  background-color: #0f77b8;
}

.c0:enabled {
  cursor: pointer;
}

<button
  class="c0"
  font-size="1"
/>
`;
